lib/api/v1/controllers/results.go

```diff
@@ -0,0 +1,15 @@
+package controllers
+
+import (
+	"github.com/gin-gonic/gin"
+)
+
+type ResultsController struct {
+	BaseController
+}
+
+func (c *ResultsController) Create(ctx *gin.Context) {
+	c.EnsureAuthenticatedUser(ctx)
```

Comment:
We can improve this as a middleware.
So this controller will be more lightweight and isolate from the authentication logic 🖖

Reply:
Added [a separate issue](https://github.com/carryall/go-google-scraper-challenge/issues/96) for this improvement 🙌🏼